src/main/groovy/nebula/plugin/dependencylock/DependencyLockTaskConfigurer.groovy

```diff
@@ -183,8 +182,8 @@ class DependencyLockTaskConfigurer {
                 }
             }
             saveTask.conventionMapping.with {
-                generatedLock = { lockfileInBuildDir }
```

Comment:
Lock file has to be evaluated lazily otherwise changes in the extension are not reflected.